version: "3"
services:

  actinia:
    image: actinia:latest
    deploy:
      replicas: 8
      resources:
        limits:
          cpus: "2"
          memory: 2048M
      restart_policy:
        condition: on-failure
    volumes:
      - ./actinia-data/grassdb:/actinia_core/grassdb:Z
      - ./actinia-data/pgpass:/mnt/pgpass:Z
      - ./actinia-data/geodata_dir:/mnt/geodata:Z
      - ./actinia-data/userdata:/actinia_core/userdata
      - ./actinia-data/workspace/temp_db:/actinia_core/workspace/temp_db
      - ./actinia-data/workspace/tmp:/actinia_core/workspace/tmp
      - ./actinia-data/resources:/actinia_core/resources
    ports:
      - "8088:8088"
    networks:
      - backend
    depends_on:
      - valkey

  valkey:
    image: valkey/valkey:8.0-alpine
    deploy:
      replicas: 1
      resources:
        limits:
          cpus: "1"
          memory: 2048M
      placement:
        constraints: [node.role == manager]
    volumes:
      - ./valkey_data:/data
    environment:
      - VALKEY_PASS_FILE=/data/config/.valkey
    command: [
      "sh", "-c",
      '
      docker-entrypoint.sh
      "/data/config/valkey.conf"
      --requirepass "$$(cat $$VALKEY_PASS_FILE)"
      '
    ]
    networks:
      - backend

networks:
  backend:
